estimatey-api/Estimatey.Infrastructure/DevOps/WorkItemSynchronizationService.cs

```diff
@@ -27,10 +27,21 @@ internal class WorkItemSynchronizationService
     }
 
     public async Task SyncProject(ProjectEntity project)
+    {
+        var workItemRevisionsBatch = await SyncProjectInternal(project);
+
+        if (!workItemRevisionsBatch.IsLastBatch)
+        {
+            Thread.Sleep(100);
```

Comment:
Also, does this run as part of Hangfire? Just wondering whether you made a conscious decision to use `Thread.Sleep` over `await Task.Delay` because the latter is non-blocking.

I'm also assuming this is here to avoid spamming API requests?